### PR TITLE
Ne 40976 properties addition valid agent config

### PR DESCRIPTION
**Pull request by Melainbal**

### CODE REVIEW

File: ne_lint/yamllint_ext/rules/inputs.py [hunk 1]
```diff
@@ -243,12 +243,7 @@ def get_type(input_obj):
 def get_default_type(input_obj):
     default_value = input_obj.default
     if type(default_value) == str:  # noqa: E721
-        if default_value.isdigit():
-            result = int
-        elif is_float(default_value):
-            result = float
-        else:
-            result = str
+        result = str
```
Comment (EarthmanT) [inline ne_lint/yamllint_ext/rules/inputs.py:246]: Do we expect if the `default_value` is a a digit like `str(123)`, then we get "str"? (same question for float? I think what we want is `int` and `float`?

File: ne_lint/yamllint_ext/rules/node_templates.py [hunk 2]
```diff
@@ -510,7 +510,9 @@ def check_terratag(model, line):
 
 
 def check_external_resource(model, line):
-    if model.is_external and 'resource_config' in model.properties:
+    if model.is_external and \
+        'resource_config' in model.properties and \
+            model.properties['use_external_resource'] == 'true':
```
Comment (EarthmanT) [inline ne_lint/yamllint_ext/rules/node_templates.py:515]: When there are more than one `and`s in a condition, in Python I like this pattern:
```
if all([
            model.is_external,
            model.properties,
            model.properties['use_external_resource'] == True
        ]):
```

Comment (Melainbal) [reply ne_lint/yamllint_ext/rules/node_templates.py]: Got it, pushed a new commit with that pattern.

File: ne_lint/yamllint_ext/rules/inputs.py (hunk 1 above)
Comment (Melainbal) [reply ne_lint/yamllint_ext/rules/inputs.py:246]: We thought about that If a user intends to use an int or float he will state it in the type of input.

File: ne_lint/yamllint_ext/autofix/colons.py [hunk 3]
```diff
@@ -9,10 +9,16 @@ def fix_colons(problem):
...
             line = lines[problem.line - 1]
+            new_line = line  # Initialize new_line with the original line
             if re.match(r'^\s*([^:\n]+)\s*:\s*([^:\n]+)\s*$', line):
                 new_line = re.sub(r'\s*:\s*', ': ', line)
             elif re.match(r'^\s*([^:\n]+)\s*:\s*$', line):
                 new_line = re.sub(r'\s*:', ': ', line)
+            elif re.match(r'^\s*([^:\n]+)\s*:\s*{.*}$', line):
+                new_line = re.sub(r'\s*:\s*', ': ', line)
+            else:
+                print(f"Line {problem.line} in file {problem.file} did not "
```
Comment (EarthmanT) [inline ne_lint/yamllint_ext/autofix/colons.py:20]: Is it possible to move this to `logger.error`?
I think the change is 
```
from ne_lint.logger import logger

....
    # when you need to log something:
    logger.error('foobar')
```

Comment (Melainbal) [reply ne_lint/yamllint_ext/autofix/colons.py]: Fixed and pushed

**Review by EarthmanT — APPROVED** [review]

**Pull request merged**